core/trino-parser/src/main/java/io/trino/sql/AstExpressionFormatter.java

```diff
@@ -0,0 +1,1288 @@
+import io.trino.sql.tree.IfExpression;
+import io.trino.sql.tree.InListExpression;
+import io.trino.sql.tree.InPredicate;
+import io.trino.sql.tree.IntervalDayTimeDataType;
+import io.trino.sql.tree.IntervalLiteral;
+import io.trino.sql.tree.IsNotNullPredicate;
+import io.trino.sql.tree.IsNullPredicate;
+import io.trino.sql.tree.JsonArray;
+import io.trino.sql.tree.JsonExists;
+import io.trino.sql.tree.JsonObject;
+import io.trino.sql.tree.JsonPathInvocation;
+import io.trino.sql.tree.JsonPathParameter;
+import io.trino.sql.tree.JsonQuery;
+import io.trino.sql.tree.JsonValue;
+import io.trino.sql.tree.LabelDereference;
+import io.trino.sql.tree.LambdaArgumentDeclaration;
+import io.trino.sql.tree.LambdaExpression;
+import io.trino.sql.tree.LikePredicate;
+import io.trino.sql.tree.LogicalExpression;
+import io.trino.sql.tree.LongLiteral;
+import io.trino.sql.tree.Node;
+import io.trino.sql.tree.NotExpression;
+import io.trino.sql.tree.NullIfExpression;
+import io.trino.sql.tree.NullLiteral;
+import io.trino.sql.tree.NumericParameter;
+import io.trino.sql.tree.OrderBy;
+import io.trino.sql.tree.Parameter;
+import io.trino.sql.tree.QualifiedName;
+import io.trino.sql.tree.QuantifiedComparisonExpression;
+import io.trino.sql.tree.Rollup;
+import io.trino.sql.tree.Row;
+import io.trino.sql.tree.RowDataType;
+import io.trino.sql.tree.SearchedCaseExpression;
+import io.trino.sql.tree.SimpleCaseExpression;
+import io.trino.sql.tree.SimpleGroupBy;
+import io.trino.sql.tree.SkipTo;
+import io.trino.sql.tree.SortItem;
+import io.trino.sql.tree.StringLiteral;
+import io.trino.sql.tree.SubqueryExpression;
+import io.trino.sql.tree.SubscriptExpression;
+import io.trino.sql.tree.SymbolReference;
+import io.trino.sql.tree.TimeLiteral;
+import io.trino.sql.tree.TimestampLiteral;
+import io.trino.sql.tree.Trim;
+import io.trino.sql.tree.TryExpression;
+import io.trino.sql.tree.TypeParameter;
+import io.trino.sql.tree.WhenClause;
+import io.trino.sql.tree.Window;
+import io.trino.sql.tree.WindowFrame;
+import io.trino.sql.tree.WindowOperation;
+import io.trino.sql.tree.WindowReference;
+import io.trino.sql.tree.WindowSpecification;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.PrimitiveIterator;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.trino.sql.AstRowPatternFormatter.formatPattern;
+import static io.trino.sql.SqlFormatter.formatName;
+import static io.trino.sql.SqlFormatter.formatSql;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+
+public final class AstExpressionFormatter
```

Comment:
Don't add the `Ast` prefix to these classes. Introduce an `Ir` prefix for the counterparts, if necessary.

None of the classes in the parser package should be touched by this change.